src/components/ShoppingListItem/ShoppingListItem.js

```diff
@@ -1,72 +1,159 @@
 import { useState, useEffect } from 'react';
 
-import isUnder24hSincePurchased from '../../utils/isUnder24hSincePurchased.js';
+import {
+  isRecentlyPurchased,
+  isPurchaseWithinUndoWindow,
+} from '../../utils/dateTimeUtils.js';
 
 import { ReactComponent as CheckboxIcon } from '../../images/icon-checkbox.svg';
 import { ReactComponent as DetailsIcon } from '../../images/icon-details.svg';
 import { ReactComponent as DeleteIcon } from '../../images/icon-delete.svg';
 import './ShoppingListItem.css';
 
-const ShoppingListItem = ({ item, checkAsPurchased, handleModalOpen }) => {
-  const [recentlyPurchased, setRecentlyPurchased] = useState(false);
+const ShoppingListItem = ({
+  item,
+  checkAsPurchased,
+  uncheckAsPurchased,
+  handleModalOpen,
+}) => {
+  const [recentlyPurchased, setIsRecentlyPurchased] = useState(false);
+  const [itemNotice, setItemNotice] = useState({});
+
+  const itemUncheckWarningMessage =
+    'You already purchased this in the last 24 hours';
+
+  // allows for undoing an accidental purchase within a certain time window
+  const handleUncheck = (item) => {
+    if (
+      item?.lastPurchaseDate?.seconds &&
+      isPurchaseWithinUndoWindow(item.lastPurchaseDate.seconds)
+    ) {
+      // if the purchase happened within the last 5 minutes, undo it
+      uncheckAsPurchased(item)
+        .then(() => {
+          setItemNotice({
+            message: `${item.itemName} purchase undone`,
+            type: 'polite',
+            error: false,
+            screenReadOnly: true,
+          });
+        })
+        .catch((err) => {
+          setItemNotice({
+            message: 'Sorry, there was a problem',
+            type: 'assertive',
+            error: true,
+            screenReadOnly: false,
+          });
+        });
+    } else {
+      setItemNotice({
+        message: itemUncheckWarningMessage,
+        type: 'assertive',
+        error: false,
+        screenReadOnly: false,
+      });
+    }
+  };
+
+  // allows for undoing an accidental purchase within a certain time window
+  const handleCheck = (item) => {
+    checkAsPurchased(item)
+      .then(() => {
+        setItemNotice({
+          message: `Purchased ${item.itemName}`,
+          type: 'polite',
+          error: false,
+          screenReadOnly: true,
+        });
+      })
+      .catch((err) => {
+        setItemNotice({
+          message: 'Sorry, there was a problem',
+          type: 'assertive',
+          error: true,
+          screenReadOnly: false,
+        });
+      });
+  };
 
   // update whether item is recently purchased
   useEffect(() => {
-    // make sure properties exist and are not null
-    if (item?.lastPurchaseDate?.seconds)
-      setRecentlyPurchased(
-        isUnder24hSincePurchased(item.lastPurchaseDate.seconds),
-      );
+    // if last purchase date is null, make sure
```

Comment:
Did you intend to leave more information here in this comment?

Reply:
@abowler2 Thanks for catching that! I will complete that thought, which I think was meant to say: 
`// if last purchase date is null, make sure to set isRecentlyPurchased to false`
I noticed that the way we were setting (or I believe not setting) isRecentlyPurchased when there was a non-null last purchase date created a bug where unchecking new items didn't work! So had to make a small change here, and wanted to note it.